cmd/devicesimulator/main.go

```diff
@@ -117,6 +136,28 @@ func main() {
 	log.Infoln("Simulator stopped.")
 }
 
+func reportVersion(versionFormat *string) {
+	cliVersion := version.Get()
+	switch *versionFormat {
+	case "":
+	case "yaml":
+		if marshalled, err := yaml.Marshal(&cliVersion); err == nil {
+			fmt.Print(string(marshalled))
+			return
+		}
+	case "json":
+		if marshalled, err := json.MarshalIndent(&cliVersion, "", "  "); err == nil {
+			fmt.Println(string(marshalled))
+			return
+		}
+	default:
+		// There is a bug in the program if we hit this case.
+		// However, we follow a policy of never panicking.
+		fmt.Printf("VersionOptions were not validated: --output=%q should have been rejected\n", *versionFormat)
```

Comment:
should this return?

Reply:
For all cases the default version will be printed. Better than nothing.

Reply:
is that true? if the user (who could be a machine) declares json returning exit code 0 with text would be quite unexpected.

Reply:
gotcha